autograder/handgrading/tests/test_views/test_handgrading_result_views.py

```diff
@@ -698,14 +698,34 @@ def test_non_staff_non_handgrader_get_handgrading_results_permission_denied(self
         self.client.force_authenticate(student)
         response = self.client.get(reverse('handgrading_results', kwargs={'pk': self.project.pk}))
         self.assertEqual(status.HTTP_403_FORBIDDEN, response.status_code)
+        
+    def test_handgrader_get_handgrading_results_only_rejected_submissions(self):
+        self.do_handgrading_results_test(self.handgrader, num_results=1, num_groups_only_reject=2, num_groups=3)
+        
+    def test_staff_get_handgrading_results_only_rejected_submissions(self):
+        self.do_handgrading_results_test(self.staff, num_results=1, num_groups_only_reject=2, num_groups=3)
 
     def do_handgrading_results_test(self, user: User, *,
-                                    num_results: int, num_groups: int = None,
+                                    num_results: int,
+                                    num_groups_only_reject: int = 0,
```

Comment:
I'd like to suggest a slightly different way of going about this: Could we change this parameter to `submission_statuses: list[ag_models.Submission.GradingStatus] | None = None`? When none, populate it as a list of "finished grading", otherwise assert its length equals num_groups. Then when creating the submissions, use the statuses from the list.

The benefit of this is that then we can test it with different non-finished-grading statuses. The important ones besides rejected are `removed_from_queue` and `error`. Let me know what you think, happy to discuss.

Reply:
So is the idea to write this method and hence other testing methods to accommodate this change?

There are currently these grading statuses
```
 class GradingStatus(models.TextChoices):
        # The submission has been accepted and saved to the database
        received = 'received'

        # The submission has been queued is waiting to be graded
        queued = 'queued'

        being_graded = 'being_graded'

        # Non-deferred test cases have finished and the group can submit
        # again.
        waiting_for_deferred = 'waiting_for_deferred'
        # All test cases have finished grading.
        finished_grading = 'finished_grading'

        # A student removed their submission from the queue before it
        # started being graded.
        removed_from_queue = 'removed_from_queue'

        # The setup command failed for an AGTestSuite marked as
        # "reject_submission_if_setup_fails".
        rejected = 'rejected'

        # Something unexpected occurred during the grading process.
        error = 'error'
```
Is the idea to accommodate for future testing?

Reply:
What do you mean by other testing methods? If possible, I'd like this PR to include tests for the `rejected`, `removed_from_queue` and `error` statuses. Let me know if it would be helpful to see some kind of example of what that would look like.

Reply:
I am wondering how we are going to test groups who do has not submitted to the AG, shall we use `list[ag_models.Submission.GradingStatus | None] | None` as the type for `submission_statuses`?

By testing methods, I meant if we are changing `do_handgrading_results_test`, we may need to rewrite how other test methods in `ListHandgradingResultsViewTestCase` call `do_handgrading_results_test`. This may take some account of work. 

For `removed_from_queue` and `error` submission, should they have the same behaviour as the `rejected` one? Do we need to test for `received`, `queued`, or `being_graded`?